### PR TITLE
Permissions

### LINKED ISSUE #84: Refresh token expiring does not push to logout screen automatically

### PR DESCRIPTION
**Pull request by harindup**
Fixes #84 and #85

### CODE REVIEW

File: frontend/src/state/types.d.ts [hunk 1]
```diff
@@ -231,7 +231,25 @@ export interface ParticipantUpdateDto {
...
+export interface HorusPermissionDto {
+    resource: HorusResource;
+    scope: HorusResourceScope;
+    type: HorusPermissionType;
+}
+
+export interface HorusAuthorityDto {
+    courseIds: number[] | null;
+    permission: HorusPermissionDto;
+}
```
Comment (JustinPraas) [inline frontend/src/state/types.d.ts]: This means that a person has HorusPermissionDto for all courses with the Ids in the list?

Comment (VriesDeRick) [reply frontend/src/state/types.d.ts]: That is correct.

Comment (JustinPraas) [reply frontend/src/state/types.d.ts]: What if a person has different HorusPermissions for different lists of courses?

Comment (harindup) [reply frontend/src/state/types.d.ts]: `HorusPermission` is atomic (not a collection). Hence, you'll simply receive a list of `HorusAuthority` which says for each `HorusPermission` what courses the user has it on.

Comment (JustinPraas) [reply frontend/src/state/types.d.ts]: Okay, that clarifies it

**Pull request merged**